### PR TITLE
lxd: always install squashfuse

### PR DESCRIPTION
**Pull request by kalikiana**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----
Fixes: [bug 730561](https://bugs.launchpad.net/snapcraft/+bug/1730561)

### CODE REVIEW

File: snapcraft/internal/lxd/_containerbuild.py [hunk 1]
```diff
@@ -223,10 +223,11 @@ def _wait_for_network(self):
         logger.info('Network connection established')
 
     def _inject_snapcraft(self):
-        if common.is_snap():
-            # Because of https://bugs.launchpad.net/snappy/+bug/1628289
-            self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
+        # Because of https://bugs.launchpad.net/snappy/+bug/1628289
+        # Needed to run snapcraft as a snap and build-snaps
+        self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
```
Comment (elopio) [inline snapcraft/internal/lxd/_containerbuild.py]: Now this is not only related to snapcraft. Shouldn't we move it to another function? `inject_snapcraft` doesn't sound like the right place for this to be.

Comment (kyrofa) [reply snapcraft/internal/lxd/_containerbuild.py]: Agreed.

Comment (sergiusens) [reply snapcraft/internal/lxd/_containerbuild.py]: @kyrofa @elopio if you guys are going to make a comment that requires a change mark the PR appropriately.

Comment (kyrofa) [reply snapcraft/internal/lxd/_containerbuild.py]: I think we both just wanted to hear justification for putting it here, not necessarily requiring a change.

Comment (kalikiana) [reply snapcraft/internal/lxd/_containerbuild.py]: I moved it to the respective container creation code paths of the two classes now.

**Pull request merged**